[docs] max-old-space-size (#52310)
elastic/kibana: docs/setup/production.asciidoc

--------

Related configurations include `elasticsearch.sniffInterval`, `elasticsearch.sniffOnStart`, and `elasticsearch.sniffOnConnectionFault`.
These can be used to automatically update the list of hosts as a cluster is resized.  Parameters can be found on the {kibana-ref}/settings.html[settings page].

[float]
[[memory]]
=== Memory

> Kibana has a default memory limit of 1.4 GB and in most cases we recommend leaving this unconfigured.  In some scenarios, such as running on containers with limited

Reviewer: ```suggestion
Kibana has a default memory limit of 1.4 GB, and in most cases, we recommend leaving this unconfigured.  In some scenarios, such as running on containers with limited
```

> Kibana has a default memory limit of 1.4 GB, and in most cases, we recommend leaving this unconfigured.  In some scenarios, such as running on containers with limited

Reviewer 2: While the max old space is generally 1.4GB, it is based on system information, like memory available. @mistic did a deep dive into this a while back and can comment.

Author: ah right, good catch

Reviewer 3: @tylersmalley exactly. Nodejs calculates the default allocated heap according multiple variables, processor architecture, total available physical memory etc (the logic can be found here https://github.com/v8/v8/blob/master/src/heap/heap.cc#L227)

So in order to be able to specify what is our minimum required memory we need to proceed what we have been discussing of distribute kibana with an override NODE_OPTIONS for heap. So, at least for now, we should not specify that information on the docs

Author: I added a clarifier,  `default _maximum_ memory limit.  it's not very specific but I'm hoping it's good enough

> , it may make sense to tweak limits to meet more specific requirements.

Reviewer 2: If you could move the comment to the previous line it will remove the space before the comma.

> This limit can be modified by setting ``--max-old-space-size` in the `NODE_OPTIONS` environment variable. For deb and rpm packages this is passed in via `/etc/default/kibana` and can be appended to the bottom of the file.

Reviewer: ```suggestion
You can modify this limit by setting ``--max-old-space-size` in the `NODE_OPTIONS` environment variable. For deb and rpm, packages this is passed in via `/etc/default/kibana` and can be appended to the bottom of the file.
```

> You can modify this limit by setting ``--max-old-space-size` in the `NODE_OPTIONS` environment variable. For deb and rpm, packages this is passed in via `/etc/default/kibana` and can be appended to the bottom of the file.

Reviewer 2: ```suggestion
You can modify this limit by setting `--max-old-space-size` in the `NODE_OPTIONS` environment variable. For deb and rpm, packages this is passed in via `/etc/default/kibana` and can be appended to the bottom of the file.
```
